portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -128,6 +132,25 @@ private Entity getCommentEntityFromComment(Comment comment) {
     return Optional.of(request.getParameter("username"));
   }
 
+  /** Returns the order for sorting comments requested by the client if it exists.
+   *  Otherwise, returns sorting by newest by default.
+   */
+  private SortOrder getSortOrder(HttpServletRequest request) {
+    if (request.getParameter("order") == null) {
+      return new SortOrder("timestamp", SortDirection.DESCENDING);
+    }
+    switch (request.getParameter("order")) {
+      case "Newest":
```

Comment:
Do you also want to handle lowercase versions of "Newest", "Oldest" and so forth? Maybe your switch statement should be `request.getParameter("order").toLowercase()`?